substrate/primitives/runtime/src/traits/transaction_extension/mod.rs

```diff
@@ -510,23 +563,20 @@ impl<Call: Dispatchable> TransactionExtension<Call> for Tuple {
 		let following_explicit_implications = for_tuples!( ( #( &self.Tuple ),* ) );
 		let following_implicit_implications = self_implicit;
 
+		let implication_parts = inherited_implication.parts();
+
 		for_tuples!(#(
 			// Implication of this pipeline element not relevant for later items, so we pop it.
 			let (_item, following_explicit_implications) = following_explicit_implications.pop_front();
 			let (item_implicit, following_implicit_implications) = following_implicit_implications.pop_front();
 			let (item_valid, item_val, origin) = {
-				let implications = (
-					// The first is the implications born of the fact we return the mutated
-					// origin.
-					inherited_implication,
-					// This is the explicitly made implication born of the fact the new origin is
-					// passed into the next items in this pipeline-tuple.
-					&following_explicit_implications,
-					// This is the implicitly made implication born of the fact the new origin is
-					// passed into the next items in this pipeline-tuple.
-					&following_implicit_implications,
-				);
-				Tuple.validate(origin, call, info, len, item_implicit, &implications, source)?
+				Tuple.validate(origin, call, info, len, item_implicit,
+					&ImplicationParts {
+						base: implication_parts.base,
+						explicit: (&following_explicit_implications, implication_parts.explicit),
+						implicit: (&following_implicit_implications, implication_parts.implicit),
```

Comment:
This keeps the order correct as  
`((Base1, Base2..), (Explicit1, Explicit2..), (Implicit1, Implicit2..))` instead of  
`((Base1, Explicit1, Implicit1), (Base2, Explicit2, Implicit2)...)`?

Reply:
On master the order is:
* For the pipeline `(A, B, C, D)` the implication in the validation of A is `(Base, ExpB, ExpC, ExpD, ImpB, ImpC, ImpD)`.
* For the pipeline `((A, B), C, D)` the implication in the validation of A is `((Base, ExpC, ExpD, ImpC, ImpD), ExpB, ImpB)`. Because when calling the validation of the inner tuple, the outer give the whole `(Base, ExpC, ExpD, ImpC, ImpD)` as inherited implication, and the inner tuple can't destructure and restructure it.

With this PR the order is in both case `(Base, ExpB, ExpC, ExpD, ImpB, ImpC, ImpD)`.